[Regression] The model-centric test fails

### What happened?

The test driver `tests/models/test_model_ops.py` now fails to run while it used to run.

### What did you expect to happen?

The driver is supposed to run op-level test cases extracted from target LLMs, such as gpt-oss-20b, as defined in yaml files under tests/resource/models/*.yaml

### How can we reproduce it?

$ pytest -n 1 --max-worker-restart=99999 -s -rsapd tests/models/test_model_ops.py

### Any environmental details we need to know?

torch-spyre: commit ac096b1c9c26ef53f624b7596ec93ce0a0ffa1ff

torch                    2.10.0

### Anything else we need to know?

_No response_

### Relevant log output

```Shell
$ pytest -n 1 --max-worker-restart=99999 -s -rsapd tests/models/test_model_ops.py
============================================================================== test session starts ==============================================================================
platform linux -- Python 3.12.12, pytest-9.0.2, pluggy-1.6.0
rootdir: /home/ohara/dt-inductor/torch-spyre
configfile: pyproject.toml
plugins: hypothesis-6.151.9, github-report-0.0.1, xdist-3.8.0, md-report-0.7.0
1 worker [0 items] / 1 error                                                                                                                                                    

==================================================================================== ERRORS =====================================================================================
________________________________________________________________ ERROR collecting tests/models/test_model_ops.py ________________________________________________________________
tests/models/test_model_ops.py:224: in <module>
    instantiate_device_type_tests(TestSpyreModelOps, globals())
../.venv/lib64/python3.12/site-packages/torch/testing/_internal/common_device_type.py:898: in instantiate_device_type_tests
    device_type_test_class: Any = type(class_name, (base, generic_test_class), {})
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   TypeError: Cannot create a consistent method resolution
E   order (MRO) for bases PrivateUse1TestBase, TestSpyreModelOps
============================================================================ short test summary info ============================================================================
ERROR tests/models/test_model_ops.py - TypeError: Cannot create a consistent method resolution
=============================================================================== 1 error in 11.64s ===============================================================================
```

--- a/tests/models/test_model_ops.py
+++ b/tests/models/test_model_ops.py
@@ -28,10 +28,10 @@ from torch.testing._internal.opinfo.core import (
     OpInfo,
 )
 from torch.testing._internal.common_device_type import (
-    PrivateUse1TestBase,
     ops,
     instantiate_device_type_tests,
 )
+from torch.testing._internal.common_utils import TestCase
 from op_registry import OP_REGISTRY, OpAdapter
 
 
@@ -110,7 +110,7 @@ _init_model_ops_db()
 seen_case_keys = set()
 
 
-class TestSpyreModelOps(PrivateUse1TestBase):
+class TestSpyreModelOps(TestCase):
     @ops(model_ops_db)
     def test_model_ops_db(
         self,
@@ -221,4 +221,4 @@ class TestSpyreModelOps(PrivateUse1TestBase):
 
 # Instantiate device type tests for the TestSpyreModelOps class
 # This is required for @ops decorator to work properly
-instantiate_device_type_tests(TestSpyreModelOps, globals())
+instantiate_device_type_tests(TestSpyreModelOps, globals(), only_for=("privateuse1",))

--- a/tests/test_spyre.py
+++ b/tests/test_spyre.py
@@ -355,6 +355,37 @@ class TestSpyre(TestCase):
         finally:
             torch.spyre.set_device(orig)
 
+    def test_instantiate_device_type_tests_mro(self):
+        """Verify that instantiate_device_type_tests works with TestCase
+        base class and only_for=("privateuse1",).
+
+        Previously, inheriting from PrivateUse1TestBase caused an MRO
+        conflict when instantiate_device_type_tests tried to create a
+        dynamic subclass that also inherits PrivateUse1TestBase.
+        Using plain TestCase + only_for avoids the conflict.
+        """
+        from torch.testing._internal.common_device_type import (
+            instantiate_device_type_tests,
+        )
+
+        class _TestMROCheck(TestCase):
+            def test_device_is_spyre(self):
+                pass
+
+        ns = {"_TestMROCheck": _TestMROCheck}
+        # This must not raise TypeError about MRO
+        instantiate_device_type_tests(_TestMROCheck, ns, only_for=("privateuse1",))
+
+        # instantiate_device_type_tests should create a class named
+        # _TestMROCheckPRIVATEUSE1 in the namespace
+        assert "_TestMROCheckPRIVATEUSE1" in ns, (
+            f"Expected _TestMROCheckPRIVATEUSE1 in namespace, got {list(ns)}"
+        )
+
+        # The generated class should be instantiable (valid MRO)
+        cls = ns["_TestMROCheckPRIVATEUSE1"]
+        assert issubclass(cls, TestCase)
+
 
 if __name__ == "__main__":
     run_tests()

